experimental/gracefulblobmissing/patches.py

```diff
@@ -238,9 +260,37 @@ def patched_loadBlob_relstorage(self, cursor, oid, serial):
         lock.close()
 
 
-def create_empty_blob(filename):
-    mkdir("-p", '/'.join(filename.split('/')[:-1]))
-    fo = open(filename, 'w')
-    fo.write("File created by experimental.gracefulblobmissing.")
-    fo.close()
-    logger.info("Created missing blob-file %s" % filename)
+def guess_extension(blob=None):
+    ''' Try to guess the type of the blob out of thin air
+    '''
+    request = getRequest()
+    if not request:
+        return 'txt'
+    try:
+        published = repr(request.PUBLISHED.im_class)
+    except:
+        published = ''
+    if 'ImageScal' in published:
+        return 'png'
+    try:
+        last_parent = repr(request.other['PARENTS'][-1])
+    except:
+        last_parent = ''
+    if 'ImageScal' in last_parent:
+        return 'png'
+    return 'txt'
+
+
+def create_empty_blob(filename, blob=None):
+    dirname = os.path.split(filename)[0]
+    if not os.path.isdir(dirname):
+        os.makedirs(dirname, 700)
```

Comment:
This gives a problem for me. The created directory does not get the requested 700, but 254 (`d-w-r-xr--`), because I have a umask of 0022. Result is that I cannot change to the directory.
This could be fixed (via [systutorials.com](https://www.systutorials.com/in-python-os-makedirs-with-0777-mode-does-not-give-others-write-permission/)):

```
oldmask = os.umask(000)
os.makedirs(directory, 0700)
os.umask(oldmask)
```

But the original code in [`ZODB/blob.py`](https://github.com/zopefoundation/ZODB/blob/1f4c6429dc9ef13f962927a50c89b46dd9a5bdee/src/ZODB/blob.py#L378-L380) simply calls `os.makedirs` so I think we should do the same.